pkg/oidc/oidc.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oidc
+
+import (
+	"fmt"
+	"net/url"
+)
+
+const (
+	defaultClientID      = "gangway"
+	defaultUsernameClaim = "email"
+	defaultGroupsClaim   = "groups"
+
+	issuerURLKey     = "--oidc-issuer-url"
```

Comment:
I'm not sure if it make sense to have `const` for those. I think the indirection reduces the readability.

Reply:
removed the constants for keys, have left the constants for default values. I think it makes sense to use constants for default values.